src/lib/db/user-store.ts

```diff
@@ -272,6 +272,10 @@ class UserStore implements IUserStore {
         await this.activeUsers().del();
     }
 
+    async deleteScimUsers(): Promise<void> {
+        await this.db(TABLE).whereNotNull('scim_id').del();
```

Comment:
normally we delete users by clearing the email and setting the deleted_at attribute. Is there  a reason to do it differently here?

Reply:
I think so, yes. If you're pushing the button that triggers this code it's because something has gone terribly wrong and you're looking for an undo button. So I've chosen to wipe them completely

Usually we'd want soft delete so we can restore or track those users later. In this flow, if you need to do that, you just your SCIM provider run again and they'll get restored, that's really the source of truth for those users anyway

I'm open to changing this though

Reply:
Hmm, if they sync again, we risk that they can get an error due to the scimExternalId already being in use.

I was thinking we might want to then remove scimExternalId and scimId as well as the email and setting the deleted_at attribute. If I remember correctly the sync for users is done by looking for user by email when configured correctly, and if the deleted_at flag is set we claim it isn't there, which will cause scim to try to write the user and then it will fail the uniqueness check, since the email has already been used?

@sighphyre ?

Reply:
I thought about this some more. I actually do want to hard delete these. @ivarconr Objections?

Reply:
We're also under the assumption that this functionality will be needed only in the case of an error configuring SCIM, so this works more or less like a DB reset, just focused on the users table. I think this should be ok... we don't want to store emails (in event logs) that were not meant to be added to the instance.